gef.py

```diff
@@ -689,10 +689,57 @@ def size(self) -> int:
             raise AttributeError
         return self.page_end - self.page_start
 
+    def _search_for_realpath_without_versions(self, path: pathlib.Path) -> str:
+        """Given a path, search for a file that exists without numeric suffixes."""
+
+        # Match the path string against a regex that will remove a suffix
+        # consisting of a dot followed by numbers.
+        candidate = re.match(r"^(.*)\.(\d*)$", str(path))
+        while candidate:
+            candidate = candidate.group(1)
+            # If the prefix from the regex match is a file, return that path.
+            if pathlib.Path(candidate).is_file():
+                return candidate
+            # Otherwise, try to match again.
+            candidate = re.match(r"^(.*)\.(\d*)$", candidate)
+        return None
+
+    def _search_for_realpath(self) -> str:
+        """This function is a workaround for gdb bug #23764
+
+        path might be wrong for remote sessions, so try a simple search for files
+        that aren't found at the path indicated, which should be canonical.
+        """
+
+        assert gef.session.remote
+        remote_path = pathlib.Path(self.path)
+        # First, try the canonical path in the remote session root.
+        candidate1 = gef.session.remote.root / remote_path.relative_to(remote_path.anchor)
+        if candidate1.is_file():
```

Comment:
`candidate1` is never re-used, could be omitted because it brings little, and the comment explains better the purpose of the test
```suggestion
        if (gef.session.remote.root / remote_path.relative_to(remote_path.anchor)).is_file():
```

Reply:
It's re-used in the return on the following line.

Reply:
(And also in the subsequent call to _search_for_realpath_without_versions()

Reply:
Ah ok, disregard, I misunderstood the function